src/values/ray-query.cxx

```diff
@@ -0,0 +1,297 @@
+    }
+
+    /// @brief Get the committed or candidate intersection object-space ray direction.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return object-space ray direction.
+    glm::vec3 getIntersectionObjectRayDirectionGLM(const unsigned intersection) const {
+        assert(intersection == 0 || intersection == 1);
+        return tlas.getIntersectionObjectRayDirection(static_cast<bool>(intersection));
+    }
+
+    /// @brief Get the committed or candidate intersection object-space ray direction.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return object-space ray direction.
+    std::vector<float> getIntersectionObjectRayDirection(const unsigned intersection) const {
+        glm::vec3 object_ray_dir = getIntersectionObjectRayDirectionGLM(intersection);
+        return std::vector<float> {object_ray_dir.x, object_ray_dir.y, object_ray_dir.z};
+    }
+
+    /// @brief Get the committed or candidate intersection object-space ray origin.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return object-space ray origin.
+    glm::vec3 getIntersectionObjectRayOriginGLM(const unsigned intersection) const {
+        assert(intersection == 0 || intersection == 1);
+        return tlas.getIntersectionObjectRayOrigin(static_cast<bool>(intersection));
+    }
+
+    /// @brief Get the committed or candidate intersection object-space ray origin.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return object-space ray origin.
+    std::vector<float> getIntersectionObjectRayOrigin(const unsigned intersection) const {
+        glm::vec3 object_ray_origin = getIntersectionObjectRayOriginGLM(intersection);
+        return std::vector<float> {object_ray_origin.x, object_ray_origin.y, object_ray_origin.z};
+    }
+
+    /// @brief Get the committed or candidate intersection 3x4 object-to-world matrix.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return 4x3 world-to-object column-major order matrix.
+    glm::mat4x3 getIntersectionObjectToWorldGLM(const unsigned intersection) const {
+        assert(intersection == 0 || intersection == 1);
+        return tlas.getIntersectionObjectToWorld(static_cast<bool>(intersection));
+    }
+
+    /// @brief Get the committed or candidate intersection 3x4 object-to-world matrix.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return 4x3 world-to-object column-major order matrix.
+    std::vector<std::vector<float>> getIntersectionObjectToWorld(const unsigned intersection) const {
+        glm::mat4x3 object_to_world = getIntersectionObjectToWorldGLM(intersection);
+
+        std::vector<std::vector<float>> result;
+        for (unsigned col = 0; col < 4; ++col) {
+            result.push_back(std::vector<float> {object_to_world[col].x, object_to_world[col].y, object_to_world[col].z}
```

Comment:
The formatting is a little wonky for this line and 291, but I suspect that using emplace_back will fix that.

Note: whenever you find yourself pushing an object you just constructed, you should probably be using emplace_back (https://en.cppreference.com/w/cpp/container/vector/emplace_back)